pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -166,7 +166,7 @@ private MetadataCacheImpl(MetadataStore store, MetadataSerde<T> serde) {
                     T newValueObj;
                     byte[] newValue;
                     try {
-                        newValueObj = modifyFunction.apply(currentValue);
+                        newValueObj = modifyFunction.apply(serde.deserialize(serde.serialize(currentValue)));
```

Comment:
Can we add a comment that explains that we are cloning the value?
Adding an explicit static method cloneValue will help in understanding better what's going on

Reply:
A comment has been added.
Now I just let the write operations of the same namespace be executed by the same thread, and will not lock, so there will be no deadlock problem